pyartcd/pyartcd/pipelines/prepare_release.py

```diff
@@ -134,6 +134,9 @@ async def run(self):
         release_config = releases_config.get("releases", {}).get(self.assembly, {})
         self.release_name = get_release_name_for_assembly(self.group_name, releases_config, self.assembly)
         self.release_version = semver.VersionInfo.parse(self.release_name).to_tuple()
+        # Convert string to an integer if necessary (like 4.20.0-0 will result in
```

Comment:
Not sure why the line above does `to_tuple()`. the semver.version.Version instance has nice attributes:

```
>>> v = semver.VersionInfo.parse('4.5.6-1')
>>> v.to_dict()
{'major': 4, 'minor': 5, 'patch': 6, 'prerelease': '1', 'build': None}
>>> v.major
4
```

But this makes me realize that the proposed versioning scheme is not semver compliant. 

Most standard compliant solution would be to have this in the build info. 

```
>>> semver.VersionInfo.parse('4.5.6-1')
Version(major=4, minor=5, patch=6, prerelease='1', build=None)
>>> semver.VersionInfo.parse('4.5.6-rc.0-0')
Version(major=4, minor=5, patch=6, prerelease='rc.0-0', build=None)
```

meaning, for an RC, semver does not understand semantically understand what is going on. 

The scheme to denote our build after a `+` seems the standard way of doing it:
```
>>> semver.VersionInfo.parse('4.5.6-rc.0+0')
Version(major=4, minor=5, patch=6, prerelease='rc.0', build='0')
>>> semver.VersionInfo.parse('4.5.6-rc.0+1')
Version(major=4, minor=5, patch=6, prerelease='rc.0', build='1')
>>> semver.VersionInfo.parse('4.5.7+1')
Version(major=4, minor=5, patch=7, prerelease=None, build='1')
```

Reply:
The doubt about the versioning scheme is resolved, `+` is not a valid token for a k8s object. We need to deviate from semver.